### DISCUSSION TITLE
Waveshare ESP32-S3-LCD-1.47 (on board ST7789 172x320)

### DISCUSSION THREAD

**Discussion by pawaller** [Bodmer/TFT_eSPI#3527] [Q&A - General] [answered]
Hi all,
Looking for any advice please on setting up one of the above with the TFT_eSPI library.
I've created a User_Setup.h from its datasheet which gives the pinouts used ...

https://www.waveshare.com/wiki/ESP32-S3-LCD-1.47?spm=a2g0o.detail.1000023.1.40eah5rHh5rHpj&file=ESP32-S3-LCD-1.47#Parameters

but it is refusing to output anything to the screen. Complete noob here so not sure if it is me.
The waveshare docs are using the lvgl library and I can compile/upload the demo and it runs fine but that seems too complicated for my needs.
Thanks in advance
Phil

**Comment by Basitadam** [comment]
Hi,
Can you send your User_Setup.h file or its contents. Also which line is uncommented in your User_Setup_Select.h file?

**Reply by pawaller** [reply]
Hi,
I uncommented the setup72 line (  #include <User_Setups/Setup72_ESP32_ST7789_172x320.h>  // Setup file for ESP32 with ST7789 1.47" 172x320 )
which seemed the closest to what I have, and modified it's contents...

// Support for 1.47" 320x172 Round Rectangle Color IPS TFT Display
#define USER_SETUP_ID 71

#define ST7789_DRIVER  // Full configuration option, define additional parameters below for this display

#define TFT_RGB_ORDER TFT_BGR  // Colour order Blue-Green-Red

#define TFT_WIDTH 172  // ST7789 172 x 320
#define TFT_HEIGHT 320  // ST7789 240 x 320

#define TFT_BL           48    // LED back-light control pin
#define TFT_BACKLIGHT_ON HIGH  // Level to turn ON back-light (HIGH or LOW)

#define TFT_MOSI 45
#define TFT_SCLK 40
#define TFT_CS   42   // Chip select control pin
#define TFT_DC   41  // Data Command control pin
#define TFT_RST  39  // Reset pin (could connect to RST pin)

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
//#define LOAD_FONT8N // Font 8. Alternative to Font 8 above, slightly narrower, so 3 digits fit a 160 pixel TFT
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

#define SPI_FREQUENCY 8000000

**Comment by Basitadam** [comment]
Can you add this into your file and try again:
```#define USE_HSPI_PORT```

**Reply by pawaller** [reply]
That did the trick!
Thank you very much :-)

Cheers

**Reply by Basitadam** [reply, marked as answer]
HappyFace();

**Reply by LeonardoDrile** [reply]
Boa tarde, Pawaller, estou enfrentando o mesmo problema com a mesma placa e estou a 2 noites literalmente sem dormir, poderia me passar seu user_setup e o user_setup_select? vai me ajudar muito.

**Reply by LucasCanete** [reply]
Having the same issue. Can we please have the user setup and user_setup_select files please?
